<h1>Python for loop vs Numpy loop<h1>

In [1]:
python_var = range(0, 1000)

In [2]:
%timeit [i**2 for i in python_var]

470 µs ± 7.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


<b> time taken for executing squares of all numbers from 0 to 1000 is 470 micro seconds

<b> Lets see for Numpy </b>

In [3]:
import numpy as np

In [4]:
numpy_var = np.arange(1000)

In [5]:
%timeit numpy_var ** 2

1.85 µs ± 48.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)


<b> time taken for numpy is 1.85 micro seconds </b>